Docs update - Expand guide for generating types from the CLI (#9820)
supabase/supabase: apps/reference/_supabase_js/working-with-types.md

## Generating types

You can use the [Supabase CLI](/docs/guides/cli) to generate types directly from your Postgres database’s schema:

> ### For databases hosted in the Supabase Cloud

Reviewer: ```suggestion
### Generate types from a remote database
```

Install the [Supabase CLI](/docs/guides/cli#installation) locally, and then run:

```bash
supabase login
supabase projects list
supabase gen types typescript --project-id <project_id> > lib/database.types.ts
```

> ### For databases running in a Supabase local development environment

Reviewer: ```suggestion
### Generate types from a local database
```

Set up your [local development environment](/docs/guides/cli/local-development), and then run:

```bash
supabase start
supabase gen types typescript --local > lib/database.types.ts
```

> For additional ways to generate types, see the [CLI Reference Docs](/docs/reference/cli/usage#supabase-gen-types).

Reviewer: ```suggestion
For additional methods, see the [CLI Reference Docs](/docs/reference/cli/usage#supabase-gen-types).
```
